[Raghava-Chittidi] iP

Raghava-Chittidi opened the pull request:
# Atlas
> “Your mind is for having ideas, not holding them.” – David Allen

Atlas frees your mind of having to remember things you need to do. It's

- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

 - [x] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Raghava-Chittidi committed:
refactor: Rename chatbot

Raghava-Chittidi committed:
feat: Add greet and exit methods

Raghava-Chittidi committed:
feat: Add echo and exit for different commands

Raghava-Chittidi committed:
feat: Add items into list

Raghava-Chittidi committed:
refactor: Move reference

Raghava-Chittidi committed:
feat: Add ability to list the items stored

Raghava-Chittidi committed:
feat: Add Task class

Raghava-Chittidi committed:
refactor: Change list to be a list of tasks

Raghava-Chittidi committed:
refactor: Modify methods to have public access modifier

Raghava-Chittidi committed:
refactor: Modify methods to have public access modifier

Raghava-Chittidi committed:
feat: Add mark and unmark commands

Raghava-Chittidi committed:
refactor: Add spacing for unmarked task

Raghava-Chittidi committed:
refactor: Modify scanner

Raghava-Chittidi committed:
refactor: Use type inferencing

Raghava-Chittidi committed:
fix: Solve merge conflicts

Raghava-Chittidi committed:
feat: Add ToDo and Deadline

Raghava-Chittidi committed:
refactor: Modify deadline to be final

Raghava-Chittidi committed:
feat: Add Event class

Raghava-Chittidi committed:
refactor: Change event toString and made Task abstract

Raghava-Chittidi committed:
feat: Add support for tracking the three types of tasks

Raghava-Chittidi committed:
refactor: Clean up comments

Raghava-Chittidi committed:
test: Add expected and actual txt files for testing

Raghava-Chittidi committed:
feat: Add exception specific for the Atlas application

Raghava-Chittidi committed:
feat: Add error handling for the various commands

Raghava-Chittidi committed:
refactor: Add methods for each command, cleaning up code in switch

Raghava-Chittidi committed:
refactor: Modify function input type

Raghava-Chittidi committed:
chore: Add additional error handling checks and modified way of splitting

Raghava-Chittidi committed:
refactor: Remove unused import

Raghava-Chittidi committed:
chore: Throw exception on unknown command

Raghava-Chittidi committed:
feat: Add support for delete task

Raghava-Chittidi committed:
fix: Resolve zero index bug and add delete task error check

Raghava-Chittidi committed:
feat: Add commands enum

Raghava-Chittidi committed:
chore: Add empty list check

Raghava-Chittidi committed:
refactor: Move files into tasks package

Raghava-Chittidi committed:
refactor: Adhere to coding standards

Raghava-Chittidi committed:
refactor: Move util functions below

Raghava-Chittidi committed:
feat: Add ability to save tasks

Raghava-Chittidi committed:
feat: Add ability to read from saved file

Raghava-Chittidi committed:
fix: Deal with unknown enum value

Raghava-Chittidi committed:
chore: Track saved tasks file

Raghava-Chittidi committed:
Merge branch 'branch-Level-7'
feat: Merge with branch-Level-7

Raghava-Chittidi committed:
feat: Add date time support for Deadline

Raghava-Chittidi committed:
feat: Add date time formatter class

Raghava-Chittidi committed:
feat: Add date time support for event and deadline

Raghava-Chittidi committed:
fix: Resolve invalid date format in atlas.txt

Raghava-Chittidi committed:
feat: Add invalid date time error handling

Raghava-Chittidi committed:
Merge branch 'branch-Level-8'
feat: Merge with branch-Level-8

Raghava-Chittidi committed:
feat: Add Storage, Parser, Ui and various commands

Raghava-Chittidi committed:
fix: Resolve ui and text ui test bugs

Raghava-Chittidi committed:
refactor: Remove wildcard import

Raghava-Chittidi committed:
Merge branch 'branch-A-MoreOOP'
Merge branch-A-MoreOOP

Raghava-Chittidi committed:
feat: Add atlas package

Raghava-Chittidi committed:
fix: Resolve runtest.sh compilation step due to new package

Raghava-Chittidi committed:
Merge branch 'branch-A-Packages'
Merge branch-A-Packages

Raghava-Chittidi committed:
Merge branch 'branch-A-Gradle'
Merge branch-A-Gradle

Raghava-Chittidi committed:
feat: Integrate gradle into project

Raghava-Chittidi committed:
Merge branch 'branch-A-Gradle'
Merge with branch-A-Gradle

Raghava-Chittidi committed:
test: Add tests for Parser, Deadline and TaskList

Raghava-Chittidi committed:
Merge branch 'branch-A-JUnit'

Raghava-Chittidi committed:
fix: Change archiveBaseName to Atlas

Raghava-Chittidi committed:
docs: Add javadoc header comments

Raghava-Chittidi committed:
feat: Add checkstyle to comply with coding standards

Raghava-Chittidi committed:
feat: Add ability to find tasks by their description

Raghava-Chittidi committed:
Merge branch 'branch-A-Javadoc'
Merge with branch-A-Javadoc

Raghava-Chittidi committed:
Merge branch 'branch-A-CodingStandard'
Merge with branch-A-CodingStandard

Raghava-Chittidi committed:
Merge with branch-Level-9

Raghava-Chittidi committed:
refactor: Comply with coding standards

Raghava-Chittidi committed:
docs: Add javadocs and comply with coding standards

LemonDrew commented on src/main/java/atlas/commands/AllCommand.java:
```diff
@@ -0,0 +1,16 @@
+    LIST,
+    MARK,
+    UNMARK,
+    TODO,
```
Perhaps a more descriptive name here? Like addTodo etc.

LemonDrew commented on src/main/java/atlas/commands/Command.java:
```diff
@@ -0,0 +1,34 @@
+     * @param storage The storage object the chatbot uses to store and load tasks
+     * @throws AtlasException The exception to be thrown in the event of any error.
+     */
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage) throws AtlasException;
```
Should the new exception be more descriptive? AtlasException doesn't seem to explain what kind of exception it is facing.

LemonDrew commented on src/main/java/atlas/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,36 @@
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws AtlasException {
+        if (this.index < 0 || this.index >= tasks.size()) {
+            throw new AtlasException("Task number does not exist.");
```
I like how you handle invalid inputs

LemonDrew commented on src/main/java/atlas/commands/MarkCommand.java:
```diff
@@ -0,0 +1,34 @@
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws AtlasException {
+        if (this.index < 0 || this.index >= tasks.size()) {
+            throw new AtlasException("Task number does not exist.");
```
I like how you handle invalid inputs.

LemonDrew commented on src/main/java/atlas/parser/Parser.java:
```diff
@@ -0,0 +1,138 @@
+                } else if (isNotNumber(commandsArray[1])) {
+                    throw new AtlasException("Task number provided is not a number.");
+                } else if (commandsArray.length > 2) {
+                    throw new AtlasException("Marking a task as done only requires the task number.");
```
Perhaps throwing or handling these exceptions would be more suitable if done inside the MarkCommand class?

LemonDrew commented on src/main/java/atlas/parser/Parser.java:
```diff
@@ -0,0 +1,138 @@
+                    throw new AtlasException("Finding tasks requires the task description.");
+                } else if (commandsArray.length > 2) {
+                    throw new AtlasException("Invalid task description.");
+                }
```
Is it much cleaner to handle all these exceptions in their corresponding classes?

LemonDrew commented on src/main/java/atlas/parser/Parser.java:
```diff
@@ -0,0 +1,138 @@
+        } catch (IllegalArgumentException e) {
+            throw new AtlasException("Unknown command.");
+        } catch (DateTimeException e) {
+            throw new AtlasException("Invalid DateTime format. Please use yyyy-mm-dd HHmm");
```
Is it better to create multiple exceptions so as to allow easier understanding of what kind of exception is being thrown?

LemonDrew commented on src/main/java/atlas/utils/DateTime.java:
```diff
@@ -0,0 +1,32 @@
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a date time class containing the methods to format date times in various formats.
```
I like how you create a separate class to handle Date and Time formatting.

LemonDrew reviewed (commented):
Overall, I find your code quite enjoyable. However, it would perhaps be better if you can further split up AtlasException into more specific ones so that others can understand what kind of error is occurring.

Raghava-Chittidi committed:
refactor: Modify method names to make it more readable

Raghava-Chittidi committed:
Merge branch 'branch-A-checkstyle'
Merge branch-A-checkstyle

Raghava-Chittidi committed:
feat: Integrate GUI and replace old UI with new GUI

Raghava-Chittidi committed:
refactor: Enable users to exit application and cleaned up unused code

Raghava-Chittidi committed:
feat: Add styling and responsiveness to GUI

Raghava-Chittidi committed:
refactor: Add javadocs and adhere to checkstyle

Raghava-Chittidi committed:
refactor: Remove unused code

Raghava-Chittidi committed:
Merge branch 'branch-Level-10'
Merge with branch-Level-10

Raghava-Chittidi committed:
chore: Add assertions to check for incorrect behaviour

The dateTime parameter passed in to the formatDate function should not be an empty string because it does not make sense to format an empty string into a LocalDate Time.

The Atlas object passed in to setAtlas in MainWindow should not be null as it will cause the application to crash

The commandsArray in the Parser's parse method should be at least of size 1 since we are going to be indexing the first element

The sections in the Storage's addTask method should be at least of size 1 since we are going to be indexing the first element

Raghava-Chittidi committed:
refactor: Remove unused import

Raghava-Chittidi committed:
refactor: Improve code quality

Refactor the Parser.parse method by moving all the sub functions to a different file as the method was getting too big.

Rename the AllCommand enum to AtlasCommand as these are commands related to the Atlas application.

Raghava-Chittidi committed:
docs: Utilise streams to find filtered tasks

Raghava-Chittidi committed:
Merge branch 'branch-A-Assertions'
Merge with branch-A-Assertions

Raghava-Chittidi committed:
chore: Resolve merge conflicts

Raghava-Chittidi committed:
Merge pull request #1 from Raghava-Chittidi/branch-A-CodeQuality

Merge with branch-A-CodeQuality

Raghava-Chittidi committed:
Merge branch 'master' into branch-A-Streams
Merge with master

Raghava-Chittidi committed:
Merge pull request #2 from Raghava-Chittidi/branch-A-Streams

Merge with branch-A-Streams

Raghava-Chittidi committed:
feat: Add github actions yml file

Raghava-Chittidi committed:
feat: Add help command functionality to the chatbot

Raghava-Chittidi committed:
feat: Add friendlier command syntax functionality

Raghava-Chittidi committed:
feat: Add more tests

Raghava-Chittidi committed:
docs: Add javadocs for test functions

Raghava-Chittidi committed:
Merge branch 'branch-A-AiAssisted'

Merge with branch-A-AiAssisted

Raghava-Chittidi committed:
fix: Resolve checkstyle issues

Raghava-Chittidi committed:
fix: Resolve checkstyle issue

Raghava-Chittidi committed:
feat: Add Atlas screenshot

Raghava-Chittidi committed:
feat: Add README information about Atlas

Raghava-Chittidi committed:
Merge branch 'branch-A-UserGuide'

Merge with branch-A-UserGuide

Raghava-Chittidi committed:
feat: Add additional README information

Raghava-Chittidi committed:
fix: Resolve GitHub actions tests failing bug

Raghava-Chittidi committed:
fix: Resolve GitHub actions tests failing bug

Raghava-Chittidi committed:
fix: Resolve GitHub actions tests failing bug

Raghava-Chittidi committed:
refactor: Remove unused imports

Raghava-Chittidi committed:
fix: Resolve GitHub actions failing tests bug

Raghava-Chittidi committed:
fix: Resolve GitHub actions failing tests bug

Raghava-Chittidi committed:
refactor: Remove unused imports

Raghava-Chittidi committed:
refactor: Resolve GitHub actions failing tests bug

Raghava-Chittidi committed:
docs: Add date time format comment

Raghava-Chittidi committed:
docs: Add exception comment

Raghava-Chittidi committed:
docs: Add storage comment

Raghava-Chittidi committed:
fix: Add event timing check

Raghava-Chittidi committed:
fix: Resolve simple command bug

Raghava-Chittidi committed:
refactor: Remove unused import
